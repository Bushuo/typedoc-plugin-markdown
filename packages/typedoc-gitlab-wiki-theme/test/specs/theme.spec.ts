import * as fs from 'fs';

import MarkdownTheme from 'typedoc-plugin-markdown/dist/theme';
import { TestApp } from 'typedoc-plugin-markdown/test/test-app';

let testApp: TestApp;

beforeAll(async () => {
  testApp = new TestApp(['theme.ts']);
  await testApp.bootstrap({ theme: './dist', entryDocument: 'home.md' });
});
describe(`Theme:`, () => {
  test(`should getUrls'`, async () => {
    const urlMappings = testApp.theme.getUrls(testApp.project);
    expect(TestApp.getExpectedUrls(urlMappings)).toMatchSnapshot();
  });
  test(`should write sidebar'`, async () => {
    const sidebarFile = fs.readFileSync(testApp.tmpobj.name + '/_sidebar.md');
    expect(sidebarFile.toString()).toMatchSnapshot();
  });
  describe(`(relativeUrls)`, () => {
    test(`should convert a url'`, async () => {
      expect(
        MarkdownTheme.HANDLEBARS.helpers.relativeURL('SomeClass.md'),
      ).toEqual('../SomeClass');
    });
    test(`should convert a url with ancor'`, async () => {
      expect(
        MarkdownTheme.HANDLEBARS.helpers.relativeURL('SomeClass.md#anchor'),
      ).toEqual('../SomeClass#anchor');
    });
  });
});
