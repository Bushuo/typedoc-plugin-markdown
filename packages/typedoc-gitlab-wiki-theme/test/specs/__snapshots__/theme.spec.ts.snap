// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme: should getUrls' 1`] = `
Array [
  "Exports.md",
  "Exports.md#declarationitema",
  "Exports.md#declarationitemb",
  "Exports.md#functionitema",
  "Exports.md#functionitemb",
  "home.md",
  "Exports.md#declarationitema",
  "Exports.md#declarationitemb",
  "Exports.md#functionitema",
  "Exports.md#functionitemb",
  "modules/Namespace:-namespaceA.md",
  "classes/Class:-SomeNestedClass.md",
  "classes/Class:-SomeNestedClass.md#constructor",
  "modules/Namespace:-namespaceB.md",
  "enums/Enumeration:-enumItemA.md",
  "enums/Enumeration:-enumItemB.md",
  "classes/Class:-ClassItemA.md",
  "classes/Class:-ClassItemA.md#constructor",
  "classes/Class:-ClassItemB.md",
  "classes/Class:-ClassItemB.md#constructor",
  "interfaces/Interface:-InterfaceItemA.md",
  "interfaces/Interface:-InterfaceItemA.md#prop",
  "interfaces/Interface:-InterfaceItemB.md",
  "interfaces/Interface:-InterfaceItemB.md#prop",
]
`;

exports[`Theme: should write sidebar' 1`] = `
"## typedoc-plugin-markdown

- [Home](home)
- [Exports](Exports)

### Namespaces

- [namespaceA](modules/Namespace:-namespaceA)
- [namespaceB](modules/Namespace:-namespaceB)

### Enumerations

- [enumItemA](enums/Enumeration:-enumItemA)
- [enumItemB](enums/Enumeration:-enumItemB)

### Classes

- [ClassItemA](classes/Class:-ClassItemA)
- [ClassItemB](classes/Class:-ClassItemB)
- [SomeNestedClass](classes/Class:-SomeNestedClass)

### Interfaces

- [InterfaceItemA](interfaces/Interface:-InterfaceItemA)
- [InterfaceItemB](interfaces/Interface:-InterfaceItemB)"
`;
